src/dataprotection/azext_dataprotection/manual/Manifests/AzureBlob.py

```diff
@@ -13,6 +13,18 @@
   "allowedRestoreTargetTypes": [ "OriginalLocation" ],
   "itemLevelRecoveyEnabled": true,
   "supportSecretStoreAuthentication": false,
+  "backupVaultPermissions": [
+    {
+      "roleDefinitionName": "Storage Account Backup Contributor",
+      "assignedOn": "storageGroup",
+      "accessMethod": [
```

Comment:
accessMethod variable looks confusing. It gives a series of variables to access to get to the right arm id but it is same for all the datasource related permission. Making it somewhat redundant and is a little cumbersome to understand too without the context. Instead, we can have a 'type' field that will denote the type of permission. For example, datasource, snapshotRG, etc. Depending on the type of the permission we can use the correct way to access the variable in the code itself. It will make the manifest more general and readable.